### ISSUE TITLE
clang-tidy 15.0.0: readability-simplify-boolean-expr

### ISSUE THREAD

**Issue by ThomasHuetsch** [llvm/llvm-project#57819]
When applying clang-tidy with the `ChainedConditionalReturn` option set to true it refactors the following code:
```c
bool foo(int& x)
{
    if (x > 0)
        return false;
    else
    {
        x = 0;
    }
    return true;
}
```

into 
```c
bool foo(int& x)
{
    return x <= 0;
}
```
The else branch is completely discarded.

**Comment by JonasToth** [comment]
this looks like a critical issue. @LegalizeAdulthood or @njames93 (i think you two know most about this check)

**Comment by LegalizeAdulthood** [comment]
I'll see if I can figure this out this weekend.... @njames93 did a big refactor recently and I wonder if this slipped in during that change.

**Comment by njames93** [comment]
This one definitely looks like something I added in

**Comment by LegalizeAdulthood** [comment]
It should be easy to figure out with a bisect.

**Comment by njames93** [comment]
https://reviews.llvm.org/D134590
With any luck we could just get this in 15.0.2 release

**Comment by LegalizeAdulthood** [comment]
Approved

**Comment by PiotrZSL** [comment]
Already fixed by 8c783b8ec78ec
